pauthabielf64/pauthabielf64.rst

```diff
@@ -542,243 +586,191 @@ by the run-time environment. There are a number of possible encoding
 schemas, each with its own trade-off. We are seeking feedback from
 platform owners about the most convenient form.
 
-Options for encoding the signing schema
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Encoding the signing schema
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-To create a signed pointer the run-time system needs to know the
-signing schema to use for the pointer. The object producer and static
-linker will need to communicate this via metadata; including at least:
+In the descriptions below the ``place`` is the operation ``P`` in
+AAELF64_ relocation descriptions. It is derived from the r_offset
+field of the relocation.
 
-* The Key, one of ``IA``, ``IB``, ``DA``, ``DB``. The ``GA`` key for
-  signing of generic data is not exposed in this ABI..
+The top 32-bits of the contents of the place is used to encode the
+signing schema for both static and dynamic relocations. This permits
+platforms using relocation compression or SHT_REL dynamic relocations
+to encode relocation addends in the bottom 32-bits. Given that the
+maximum size of link-units using the small code-model is 4 gigabytes
+this should be sufficient.
 
-* The constant discriminator value.
+.. class:: pauthabiencoding
 
-* Whether to combine address diversity with the discriminator.
++------+-------------------+----------+----------+----------+---------------+---------------------+
+| 63   | 62                | 61       | 60:59    | 58:48    |  47:32        | 31:0                |
++======+===================+==========+==========+==========+===============+=====================+
+| auth | address diversity | reserved | key      | reserved | discriminator | reserved for addend |
```

Comment:
Should the specification require these reserved bits to be set to 0 and recommend that consumers of the field require the reserved bits to be set to 0? This should allow them to be safely repurposed later.

Reply:
That seems reasonable, I expect that in the early stages we'll be recompiling everything but it won't hurt to be more specific. I'll add some text.